gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -504,6 +506,189 @@ def _get_variant_calls(self, record, formats):
     return calls
 
 
+class PySamParser(VcfParser):
+  """An Iterator for processing a single VCF file using PySam.
+
+  PySam allows reading a file through either stdin stream, or through as actual
+  VCF files, for which it requires legitimate file descriptor. Since we want to
+  perform our own parallelization, we will fork our process, to use 2 pipelines
+  that will feed into PySam object - 1 will feed the data from main process
+  throughout into the child one, while second will get that data in child thread
+  and feed it to PySam library.
+
+  The requirement of using two pipelines comes from the design of VcfParser base
+  class - we could only use a single pipe, but it will divert the parsers.
+  """
+
+  def __init__(self,
+               file_name,  # type: str
```

Comment:
python style question: we don't line these up we leave them jagged? (the type annotations)